app/src/main/java/ch/epfl/sdp/blindly/UserMapActivity.kt

```diff
@@ -35,8 +54,56 @@ class UserMapActivity: AppCompatActivity(), OnMapReadyCallback, ActivityCompat.O
         val mapFragment = supportFragmentManager
                 .findFragmentById(R.id.map) as SupportMapFragment?
         mapFragment!!.getMapAsync(this)
+
+        // Cancel loading if we can't get the user id
+        val currentUserId = userHelper.getUserId() ?: return
+        val matchId = intent.extras?.getString(MATCH_ID) ?: "default_user"
+        matchName = intent.extras?.getString(MATCH_NAME) ?: getString(R.string.default_label_map_pin)
+
+        locationDatabase = databaseHelper.getLocationLiveDatabase(currentUserId, matchId)
+        locationDatabase.addListener(object :
+            DatatbaseHelper.BlindlyLiveDatabase.EventListener<BlindlyLatLng>() {
+            override fun onMessageReceived(message: Message<BlindlyLatLng>) {
+                updateMarker(message)
+            }
+            override fun onMessageUpdated(message: Message<BlindlyLatLng>) {
+                updateMarker(message)
+            }
+        })
+
+        AndroidLocationService(applicationContext).addLocationChangeListener( object :
+            AndroidLocationService.LocationChangeListener() {
+            override fun onLocationChange(pos: BlindlyLatLng) {
+                locationDatabase.updateLocation(pos)
+            }
+        })
+    }
+
+    private fun updateMarker(message: Message<BlindlyLatLng>) {
+        if (!::map.isInitialized) return
+        if (message.currentUserId == userHelper.getUserId()) return
+        val pos = message.messageText ?: return
+        val latLng = pos.toLatLng() ?: return
+        tryToAddMarker()?.position = latLng
+        // Always keep match position on the center of the map
+        map.moveCamera(CameraUpdateFactory.newLatLng(latLng))
```

Comment:
Isn't it a better idea to keep ourselves at the center of the map ?

Reply:
Yep, that's a good idea ! This way the app can show us where to go at the beginning, and then we're free to move the map.

Reply:
Yes, the pins are different.